src/App.jsx

```diff
@@ -14,45 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
+
+const compareGoods = (good1, good2, orderBy) => {
+  switch (orderBy) {
+    case SORT_BY_NAME:
+      return good1.localeCompare(good2);
+    case SORT_BY_LENGTH:
+      return good1.length - good2.length;
+
+    default:
+      return 0;
+  }
+};
+
+const getPreparedGoods = (goods, { sortBy, query, reverse }) => {
+  const filtered = query
+    ? [...goods].filter(good => good.include(query))
+    : [...goods];
+  const sorted = sortBy
+    ? filtered.sort((good1, good2) => compareGoods(good1, good2, sortBy))
+    : filtered;
+
+  return reverse
+    ? sorted.reverse()
+    : sorted;
+};
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState(null);
+  const [reverse, setReverse] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Reply:
Done